src/pages/create.tsx

```diff
@@ -1,11 +1,228 @@
-import React, { ReactElement } from "react";
-import { Button } from "react-bootstrap";
+import React, { FormEvent, ReactElement, useState } from "react";
+import { Button, Card, Col, Form, FormControl, InputGroup, Row } from "react-bootstrap";
+import { useSelector } from "react-redux";
+import { AppState, FormControlElement } from "../lib/types";
+import { TFunction } from "next-i18next";
+import ConnectButton from "../components/connect-button/connect-button";
+import SpotPrice from "../components/spot-price/spot-price";
+import { withTranslation } from "../common/i18n";
+import { encodeOptionData, hoursToMs, daysToMs } from "../common/utils";
+import { Option } from "../lib/types/option";
+
+const VALIDITY_OPTIONS = {
+    hours: [6, 12],
+    days: [1, 2],
+};
+
+enum SubmitStates {
+    None,
+    Processing,
+    Success,
+    Failure,
+}
+
+const Create = ({ t }: { readonly t: TFunction }): ReactElement => {
+    const isConnected = useSelector((state: AppState) => state.user.isConnected);
+
+    const defaultState: Option = {
```

Comment:
I feel like keeping this sort of thing out of the view is cleaner.
For `User` I did this: https://github.com/interlay/xoptsv3-ui/blob/dev/src/lib/entities/user.ts

The idea is to keep the `types` for type only things (e.g. `Optional`) and have what would be the model in a typical MVC in `entities`.
Let me know what you think.

Reply:
Definitely good call, I didn't think of it while building but will update in the morning